docs/src/languages/php/tuning.md

```diff
@@ -3,216 +3,205 @@ title: "Performance tuning"
+5. Check the OPcache status for FastCGI commands by running the following command:
 
-The timestamp validation may be disabled with an `ini` setting,
-and the easiest way to do so is via `.platform.app.yaml`:
+   ```bash
+   php cachetool.phar opcache:status --fcgi=$SOCKET
+   ```
 
-```yaml
+   The `--fcgi=$SOCKET` option ensures the PHP-FPM process on the server connects through the right socket.
+6. Analyze the output to determine the optimal value for `opcache.memory_consumption`.
+   The most important values from CacheTool's output are:
+
+   - `Memory used`
+   - `Memory free`
+   - `Oom restarts` (out of memory restarts)
+     If the value is different than 0, you don't have enough memory allocated to OPcache.
+
+   If `Memory free` is too low or `Oom Restarts` too high,
+   set a higher value for memory consumption.
+7. Set `opcache.memory_consumption`.
+   Note: The unit for `opcache.memory_consumption` is megabytes.
+
+   An example configuration:
+
+   ```yaml {location=".platform.app.yaml"}
+   variables:
+       php:
+           'opcache.memory_consumption': 96
+   ```
+
+8. [Restart PHP-FPM](#restart-php-fpm) and make sure that OPcache works as expected by rerunning CacheTool with:
+
+    ```bash
+    php cachetool.phar opcache:status --fcgi=$SOCKET
+    ```
+
+9. Remove CacheTool by deleting the `cachetools.phar` file with `rm -rf cachetools.phar`.
+
+### Disable OPcache timestamp validation
+
+By default, OPcache checks that the cached version of a file is always up-to-date.
+This means that every time a cached file is used, OPcache compares it to the file on disk.
+If that file has changed, it gets reloaded and re-cached.
+This allows to support apps that generate compiled PHP code from user configuration.
+
+If you know your code isn't going to change outside a new deploy,
+you can disable that check and get a small performance improvement.
+
+Timestamp validation can be disabled by adding the following variable to your [app configuration](../../create-apps/_index.md):
+
+```yaml {location=".platform.app.yaml"}
 variables:
     php:
         'opcache.validate_timestamps': 0
 ```
 
+When you have disabled OPcache timestamp validation,
+you need to explicitly clear OPcache on deployment by [restarting PHP-FPM](#restart-php-fpm).
+
+Note: If your app generates PHP code at runtime based on user configuration, don't disable timestamp validation.
+Doing so would prevent updates to the generated code from being loaded.
+
+## Restart PHP-FPM
+
+To force a restart of PHP-FPM:
+1. Connect to the container via SSH using the [CLI](../../administration/cli/_index.md) by running `platform ssh`
+2. Run `sv restart app`
+
+To force a restart of PHP-FPM on every deployment, add the `sv restart app` command to your [`deploy` hook](../../create-apps/hooks/hooks-comparison.md#deploy-hook).
+
+To clear the cache when PHP-FPM is restarted, add the `sv restart app` command to your [`start` command](../../create-apps/app-reference.md#web-commands).
```

Comment:
The issue here is if we move the container to a different host, the deployment hook isn't run.
So to refresh the cache, without having to do a manual redeployment, it seems good practice to always add the restart.

What would help clarify that?
Should I provide more context? Move it to "Disable OPcache timestamp validation"?

Reply:
It might help if the command weren't repeated because the word order seems currently to emphasize it. But it can probably survive either way.
```suggestion
To force a restart of PHP-FPM on every deployment, add this command to your [`deploy` hook](../../create-apps/hooks/hooks-comparison.md#deploy-hook).

To clear the cache when PHP-FPM is restarted, add this command to your [`start` command](../../create-apps/app-reference.md#web-commands).
```